plutus-core/plutus-core/src/PlutusCore/Rename.hs

```diff
@@ -60,11 +66,17 @@ instance Rename a => Rename (Normalized a) where
 -- 2. some value may be used multiple times
 --
 -- so we annotate such a value with 'Dupable' and call 'liftDupable' at each usage, which ensures
--- global conditions is preserved.
+-- global uniqueness is preserved.
+--
+-- 'unDupable' is not supposed to be exported. Don't provide any instances allowing the user to
+-- access the underlying value.
 newtype Dupable a = Dupable
```

Comment:
I wonder if you want to define this next to `Renamed` and discuss them together. They have sort of modal logic relationship:
- `Renamed` == necessarily globally unique
- `Dupable` == not necessarily globally unique == possibly not globally unique

Maybe they could have more symmetrical names too. I don't have great suggestions ("Renamed" is a bit about what's *happened* to it not the *property*). "GloballyUnique"/"NonGloballyUnique"?